### PR TITLE
chore(git): ignore dist files until release

### PR DESCRIPTION
**Pull request by blake-newman**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

# Reasons
ignore dist files until release
improves contributor experience
ensures dist files can't be accidentally committed in

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,9 @@ test/e2e/reports
 test/e2e/screenshots
 coverage
 RELEASE_NOTE*.md
+dist/*.js
+packages/vue-server-renderer/basic.js
+packages/vue-server-renderer/build.js
+packages/vue-server-renderer/server-plugin.js
+packages/vue-server-renderer/client-plugin.js
+packages/vue-template-compiler/build.js
```
Comment (nickmessing) [inline .gitignore]: Why did you remove `\n` from the end? `cat` users will hate you :smile:

Comment (blake-newman) [inline .gitignore]: Good catch

**Comment by blake-newman** [comment]
@yyx990803 Ill put this across all the core repos with dist folders in coming days.
